squirrel.frontier/src/test/java/org/dice_research/squirrel/predictor/impl/PredictorImplTest.java

```diff
@@ -0,0 +1,181 @@
+import java.net.URISyntaxException;
+
+import static org.junit.Assert.assertEquals;
+
+public class PredictorImplTest {
+    private CrawleableUri curi;
+    private PredictorImpl predictor;
+
+    @Test
+    public void train() throws Exception {
+        // //Initialization
+        curi = new CrawleableUri(new URI("https://mcloud.de/web/guest/suche/-/results/search/55?_mcloudsearchportlet_sort=latest"));
+        CrawleableUri curiPos = new CrawleableUri(new URI("https://mcloud.de/export/datasets/037388ba-52a7-4d7e-8fbd-101a4202be7f"));
+        CrawleableUri curiNeg = new CrawleableUri(new URI("https://ckan.govdata.de"));
+
+        predictor = new PredictorImpl();
+        predictor.TRAINING_SET_PATH = "trainDataset.txt";
+
+        // train the learner on two URIs: one RDF and one non RDF
+        predictor.train();
+
+        // predict for a random URI(HTML) example
+        predictor.featureHashing(curi);
+        double pred = predictor.predict(curi);
+        double pround = Math.round(pred*100.0)/100.0;
+        Assert.assertEquals(0.41,pround,0.1);
+
+        // predict for a Positive (RDF) example
+        predictor.featureHashing(curiPos);
+        double pred2 = predictor.predict(curiPos);
+        double pround2 = Math.round(pred2*100.0)/100.0;
+        Assert.assertEquals(0.45, pround2,0.1);
+
+        // predict for a negative (non RDF) example
+        predictor.featureHashing(curiNeg);
+        double pred3 = predictor.predict(curiNeg);
+        double pround3 = Math.round(pred3*100.0)/100.0;
+        Assert.assertEquals(0.43, pround3,0.1);
+    }
+
+    @Test
+    public  void evaluation() throws  Exception{
+        predictor = new PredictorImpl();
+        predictor.TRAINING_SET_PATH = "predictor/trainData.txt";
+        Integer uriCount = 0;
+        Integer correctCount = 0;
+        double accuracy;
+        // train the learner on two URIs: one RDF and one non RDF
+        predictor.train();
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(getClass().getClassLoader().getResourceAsStream("predictor/evalDataSet.txt")) ))       {
+            String line;
+            while ((line = br.readLine()) != null) {
+                uriCount ++;
+                String[] split = line.split(",");
+                URI furi = null;
+                try {
+                    furi = new URI(split[1]);
+                } catch (URISyntaxException e) {
+                    e.printStackTrace();
+                }
+                CrawleableUri uri = new CrawleableUri(furi);
+                predictor.featureHashing(uri);
+                double pred = predictor.predict(uri);
+                if(split[0].equals("RDF")){
+                    if(pred >= 0.5){
+                        correctCount ++;
+                    }
+                }
+                else{
+                    if(pred < 0.5){
+                        correctCount ++;
+                    }
+                }
+            }
+        }
```

Comment:
You are implementing the same code of reading a CSV file again although you already have it in the PredictorImpl class. You should try to never repeat yourself :wink: